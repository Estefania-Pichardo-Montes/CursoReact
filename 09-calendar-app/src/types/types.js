
export const types = {

    uiOpenModal: '[ui] Open modal',
    uiCloseModal: '[ui] Close modal',


    eventSetActive: '[event] Set Active',
    eventLogout:'[event] Logout event',

    eventStartAddNew:'[events] start Add New',
    eventAddNew: '[event] Add new',
    eventClearActiveEvent: '[event] Clear active event',
    eventUpdated: '[event] Event updated',
    eventDeleted: '[event] Event deleted',
    eventLoaded:'[event] Events Loaded',

    authCheckingFinish: '[auth] Finish checking Login State',
    authStartLogin:'[auth] Start Login',
    authLogin:'[auth] Login',
    authStartRegister:'[auth] Start Register',
    authStartTokenRenew:'[auth] Start Token Renew',
    authLogout:'[auth] Logout'



}