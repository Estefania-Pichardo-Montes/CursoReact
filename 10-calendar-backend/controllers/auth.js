const {response, json, request}=require('express');
const Usuario = require('../models/Usuario');
const bcrypt=require('bcryptjs');
const {generarJWT} = require('../helpers/jwt');


const crearUsuario= async(req, res= response)=>{
    const { email,password}=req.body;
    

    try{

        //Validacion de que no exista el usuario
        let usuario= await Usuario.findOne({email});
        if(usuario){
            return res.status(400).json({
                ok:false,
                msg:'Ya existe un usuario con ese correo'
            })
        }

        //Crear una nueva instancia del usuario
        usuario = new Usuario(req.body);

        //Encriptar contraseña
        const salt=bcrypt.genSaltSync();
        usuario.password=bcrypt.hashSync(password,salt);

        //Guardamos el usuario en la BD
        await usuario.save();

        //Generar Json Web Token
        const token= await generarJWT(usuario.id, usuario.name);


        res.status(201).json({
            ok:true,
            uid:usuario.id,
            name:usuario.name,
            token:token
        });


    }catch(error){
        console.log(error);
        res.status(500).json({
            ok:false,
            msg:'Por favor hable con el administrador'
        })
    }
    
}

const loginUsuario= async(req, res=response)=>{

    const {email,password}=req.body;

    try{

        const usuario= await Usuario.findOne({email});
        if(!usuario){
            return res.status(400).json({
                ok:false,
                msg:'El usuario no existe con ese email'
            });
        }
        //Confirmar los passwords
        const validPassword=bcrypt.compareSync(password,usuario.password)

        if(!validPassword){
            return res.status(400).json({
                ok:false,
                msg:'Password incorrecto'
            });
        }
        //Si las contraseñas coinciden generar el web token
        const token= await generarJWT(usuario.id, usuario.name);


        res.json({
            ok:true,
            uid:usuario.id,
            name:usuario.name,
            token:token
        })

    }catch(error){

        console.log(error);
        res.status(500).json({
            ok:false,
            msg:'Por favor hable con el administrador'
        })
    }
}


const revalidarToken= async(req, res=response)=>{

    const {uid,name}=req

    //Generar un nuevo JWT u retornarlo en esta peticion
    const token=await generarJWT(uid,name);

    res.json({
        ok:true,
        uid,
        name,
        token
       
    })
}

//Exportamos las funciones
module.exports={
    crearUsuario,
    loginUsuario,
    revalidarToken

}