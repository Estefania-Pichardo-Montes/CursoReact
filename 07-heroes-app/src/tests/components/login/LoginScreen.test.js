import React from 'react';
import {mount} from 'enzyme'
import { AuthContext } from '../../../auth/AuthContext';
import { LoginScreen } from '../../../Components/login/LoginScreen';
import { types } from '../../../types/types';

describe('Pruebas en Login screen', () => {
    
    const history={
        replace:jest.fn()
    }
  
    const contextValue={
        dispatch:jest.fn(),
        user:{
            logged:false
        }
    }
    const wrapper=mount(

        <AuthContext.Provider value={contextValue}>
            <LoginScreen history={history}/>
        </AuthContext.Provider>
    )

    test('Debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar la navegacion', () => {

        const handleClick=wrapper.find('button').prop('onClick');
        handleClick();

        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type:types.login,
            payload:{
                name:'Estefania'
            }
        });
        expect(history.replace).toHaveBeenCalledWith('/');

        localStorage.setItem('lastPath','/dc');
        handleClick();
        expect(history.replace).toHaveBeenCalledWith('/dc');
    })
    
    
})
