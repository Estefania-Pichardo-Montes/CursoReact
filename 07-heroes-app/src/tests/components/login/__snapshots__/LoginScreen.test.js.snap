// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Login screen Debe de mostrar el componente correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
